coalib/bearlib/spacing/SpacingHelper.py

```diff
@@ -86,11 +86,28 @@ def replace_spaces_with_tabs(self, line: str):
         :param line: The string with spaces to replace.
         :return:     The converted string.
         """
+        def previous_whitespace():
+            # Find the previous real character, and remaining chars to fill tab
+            non_whitespace_position = tabless_position - currspaces
+            tab_fill = non_whitespace_position % self.tab_width
+
+            if tab_fill and tab_fill + currspaces >= self.tab_width:
+                whitespace = "\t"
+                remaining_spaces = currspaces - (self.tab_width - tab_fill)
+            else:
+                whitespace = ""
+                remaining_spaces = currspaces
+
+            whitespace += "\t" * (remaining_spaces // self.tab_width)
+            whitespace += " " * (remaining_spaces % self.tab_width)
+            return whitespace
+
         currspaces = 0
         result = ""
         # Tracking the index of the string isnt enough because tabs are
         # spanning over multiple columns
         tabless_position = 0
+        previous_char = None
```

Comment:
you use it only for appending to a string, appending `None` raises an error, so there's imo no meaning to it setting it to `None` beforehand.